command/var_put.go

```diff
@@ -0,0 +1,472 @@
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/mapstructure"
+	"github.com/posener/complete"
+)
+
+type VarPutCommand struct {
+	Meta
+
+	contents  []byte
+	inFmt     string
+	outFmt    string
+	tmpl      string
+	testStdin io.Reader // for tests
+}
+
+func (c *VarPutCommand) Help() string {
+	helpText := `
+Usage: nomad var put [options] <path> [<key>=<value>]...
+
+  The 'var put' command is used to create or update an existing variable.
+  The items to be stored in the variable can be supplied as a series of
+  key-value pairs. The value for a key-value pair can be a string, an
+  @-prefixed file reference, or a '-' to get the value from standard input.
```

Comment:
We're not really documenting that you can pass the entire variable in from file/stdin, only that you can pass in individual key values from string/file/stdin. i.e. that both `nomad var put @myfile.json` and `nomad var put some/path user=me password@mypassword.txt` work. You can kind of pick it up from the `-in` flag below but probably better to be explicit about that here.

Reply:
Took a swing at being more explicit here. Please give it a new read.